imap/index.c

```diff
@@ -6132,6 +6133,32 @@ MsgData **index_msgdata_load(struct index_state *state,
     if (found_anchor)
         *found_anchor = 0;
 
+    /* set mailbox level states */
+    for (j = 0; sortcrit[j].key; j++); // count how many we need
+    preload = xzmalloc(j * sizeof(int));
+    for (j = 0; sortcrit[j].key; j++) {
+        label = sortcrit[j].key;
+        switch(label) {
+        case SORT_SAVEDATE:
+#ifdef WITH_JMAP
+        case SORT_SNOOZEDUNTIL:
+#endif
+            preload[j] = !strcmpnull(mailbox_uniqueid(mailbox), sortcrit[j].args.mailbox.id);
+            break;
+
+        case SORT_HASCONVFLAG:
+            preload[j] = -1;
+            if (!cstate) cstate = conversations_get_mbox(index_mboxname(state));
+            assert(cstate);
```

Comment:
Are we sure that this code is not reachable for builds without conversations? The alternative would be to gracefully not preload if cstate is null.

Reply:
A little later (inside the loop):

```
            if ((label == SORT_HASCONVFLAG || label == SORT_CONVMODSEQ ||
                label == SORT_CONVEXISTS || label == SORT_CONVSIZE) && !did_conv) {
                if (!cstate) cstate = conversations_get_mbox(index_mboxname(state));
                assert(cstate);
```

So if so, this isn't a regression.

Reply:
ack